### PR TITLE
Make RyuJIT/arm32 the default arm32 JIT

### PR DESCRIPTION
**Pull request by BruceForstall**
RyuJIT/arm32 JIT is now named clrjit.dll (was named protojit.dll).

arm32 LEGACY_BACKEND JIT is now named legacyjit.dll (was named clrjit.dll).

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet-bot test CI please

**Comment by BruceForstall** [comment]
@dotnet/arm32-contrib @dotnet/jit-contrib

File: netci.groovy [hunk 1]
```diff
@@ -1603,10 +1603,10 @@ def static calculateBuildCommands(def newJob, def scenario, def branch, def isPR
 
                     def buildArchitecture = 'arm'
 
-                    // For 'arm' (the RyuJIT/arm32 architecture), tell build.cmd to use RyuJIT/arm32 for crossgen compilation.
-                    // RyuJIT/arm32 is currently not the default JIT; it is an aljit. So, this is a special case.
+                    // For 'armlb' (the RyuJIT/arm32 architecture), tell build.cmd to use RyuJIT/arm32 for crossgen compilation.
+                    // Legacy backedn is not the default JIT; it is an aljit. So, this is a special case.
```
Comment (Suchiman) [inline netci.groovy]: typo `backedn`

**Comment by BruceForstall** [comment]
@dotnet-bot test CI please

**Comment by BruceForstall** [comment]
Timeout?
```21:53:31  > git fetch --tags --progress https://github.com/dotnet/dotnet-ci +refs/heads/*:refs/remotes/origin/* --depth=1 # timeout=30
21:53:31 ERROR: Error fetching remote repo 'origin'
21:53:31 
hudson.plugins.git.GitException: Failed to fetch from https://github.com/dotnet/dotnet-ci
21:53:31 	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:817)
21:53:31 	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1084)
21:53:31 	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1115)
21:53:31 	at org.jenkinsci.plugins.multiplescms.MultiSCM.checkout(MultiSCM.java:143)
21:53:31 	at hudson.scm.SCM.checkout(SCM.java:495)
21:53:31 	at hudson.model.AbstractProject.checkout(AbstractProject.java:1212)
21:53:31 	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:566)
21:53:31 	at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
21:53:31 	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:491)
21:53:31 	at hudson.model.Run.execute(Run.java:1737)
21:53:31 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
21:53:31 	at hudson.model.ResourceController.execute(ResourceController.java:97)
21:53:31 	at hudson.model.Executor.run(Executor.java:421)
21:53:31 Caused by: hudson.plugins.git.GitException: Command "git fetch --tags --progress https://github.com/dotnet/dotnet-ci +refs/heads/*:refs/remotes/origin/* --depth=1" returned status code 128:
21:53:31 stdout: 
21:53:31 stderr: fatal: git fetch-pack: expected shallow list
21:53:31 
```
https://ci.dot.net/job/dotnet_coreclr/job/master/job/GenPRTest/job/generator_prtest/685/

try again.

@dotnet-bot test CI please

**Comment by alpencolt** [comment]
@BruceForstall what about `armelnonjit`, `protononjit` and other subprojects?
Are they needed changes?

File: netci.groovy [hunk 2]
```diff
@@ -1574,10 +1574,10 @@ def static calculateBuildCommands(def newJob, def scenario, def branch, def isPR
 
                     def buildArchitecture = 'arm'
 
-                    // For 'arm' (the RyuJIT/arm32 architecture), tell build.cmd to use RyuJIT/arm32 for crossgen compilation.
-                    // RyuJIT/arm32 is currently not the default JIT; it is an aljit. So, this is a special case.
+                    // For 'armlb' (the RyuJIT/arm32 architecture), tell build.cmd to use RyuJIT/arm32 for crossgen compilation.
```
Comment (hqueue) [inline netci.groovy]: Nit. misleading comments.
Please update comment something like ` // For 'armlb' (the legacy backend for arm), tell build.cmd to use legacy backend for crossgen compilation.`

**Comment by BruceForstall** [comment]
> @BruceForstall what about armelnonjit, protononjit and other subprojects?
Are they needed changes?

These still make sense to keep: they are x86 hosted, arm targeting altjits, and they are not replaced by any other existing build. The "protojit" directory is now unused, although our internal builds still have some pointers to it, so I'll remove it later.

File: netci.groovy (hunk 2 above)
Comment (BruceForstall) [reply netci.groovy]: Fixed

**Comment by BruceForstall** [comment]
I'm planning to disable existing failing tests (just 3 in various stress modes), updating this change if necessary, and merging this, possibly today.

**Comment by BruceForstall** [comment]
@dotnet-bot test CI please

**Comment by BruceForstall** [comment]
@jashook @RussKeldorph @dotnet/arm32-contrib @dotnet/jit-contrib PTAL

**Pull request merged**